### DISCUSSION TITLE
How is the actual XML fed into Task Scheduler?

### DISCUSSION THREAD

**Discussion by ddDerelict** [dahall/TaskScheduler#986] [Q&A]
I'm able to import an XML file into Task Scheduler using
TaskService.Instance.RootFolder.ImportTask("taskName", "filePath");
;
However, with
// Create a Task using XML with user information
TaskService.Instance.RootFolder.RegisterTask("NewTask1", xml, TaskCreation.Create,
   "SYSTEM", null, TaskLogonType.SystemAccount);

how is Task Scheduler getting the actual XML in order to register a new task?

**Comment by dahall** [comment]
The `RegisterTask` method takes a `string` that contains all the XML in the same format as the contents of the file you are feeding into `ImportTask`.

**Reply by ddDerelict** [reply]
Apologies for delayed response - had stopped monitoring regularly ...

> RegisterTask method takes a string that contains all the XML

That's what I thought so with a StringBuiloder/XmlWriter: approach
```
StringBuilder xmlSB = new StringBuilder();
using (XmlWriter w = XmlWriter.Create(xmlSB, wSettings)) {define xml}
TaskService.Instance.RootFolder.RegisterTask($"{taskName}", xmlSB.ToString(),
    TaskCreation.Create,"SYSTEM", null, TaskLogonType.SystemAccount);

```
I have an app (C#/Winforms) supporting cyclical preventative maintenance which registers a reminder in Task Scheduler; task registration currently requires running elevated as Admin. I was hoping the 

> ""SYSTEM", null, TaskLogonType.SystemAccount"

part of the command would enable task registration without the user needing to have Admin privileges but I get a compiler error:
 "TaskLogonType does not contain a definition for SystemAccount.".

Is SYSTEM in the command not referencing the Windows built-in System account or am I misinterpreting this?

**Reply by dahall** [reply]
That is the correct syntax. See [help docs](https://dahall.github.io/TaskScheduler/html/M_Microsoft_Win32_TaskScheduler_TaskFolder_RegisterTask.htm) for more detail.
